ui/v2.5/src/components/Scenes/SceneDetails/RatingSystem.tsx

```diff
@@ -0,0 +1,363 @@
+import React, { useState, useRef } from "react";
+import * as GQL from "src/core/generated-graphql";
+import { ConfigurationContext } from "src/hooks/Config";
+import Box from "@mui/material/Box";
+import StarIcon from "@mui/icons-material/Star";
+import StarBorderIcon from "@mui/icons-material/StarOutline";
+import { FormattedMessage, useIntl } from "react-intl";
+import { Col, Form, Row } from "react-bootstrap";
+import { FormUtils } from "src/utils";
+
+export interface IRatingSystemProps {
+  value?: number;
+  onSetRating?: (value?: number) => void;
+  disabled?: boolean;
+  excludeLabel?: boolean;
+}
+
+export interface IRatingStarsProps {
+  value?: number;
+  onSetRating?: (value?: number) => void;
+  disabled?: boolean;
+  precision: number;
+  maxRating: number;
+}
+
+export interface IRatingNumberProps {
+  value?: number;
+  onSetRating?: (value?: number) => void;
+  disabled?: boolean;
+}
+
+export const RatingNumber: React.FC<IRatingNumberProps> = (
+  props: IRatingNumberProps
+) => {
+  const [input, setInput] = useState<string | "0.0">();
+  const [previous, setPrevious] = useState<string | "0.0">();
+  const [useValidation, setValidation] = useState<boolean | true>();
+  function stepChange() {
+    setValidation(false);
+  }
+
+  function nonStepChange() {
+    setValidation(true);
+  }
+
+  function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
+    let val = e.target.value;
+    if (!useValidation && props.onSetRating != null) {
+      e.target.value = Number(val).toFixed(1);
+      setInput(Number(val).toFixed(1));
+      setPrevious(Number(val).toFixed(1));
+      let tempVal = Number(val) * 10;
+      props.onSetRating(tempVal != 0 ? tempVal : undefined);
+      setValidation(true);
+      return;
+    }
+    const match = /(\d{0,1})(\d{0,1})(.?)((\d{0,1})?)/g.exec(val);
+    const matchOld = /(\d{0,1})(\d{0,1})(.?)((\d{0,2})?)/g.exec(previous ?? "");
```

Comment:
This one simplified to `(\d)(\d)(.?)((\d{0,2})?)`

Reply:
On a side note, it seems like it might have been easier to just convert the string to a number and validate the number instead of the string. This would also help the readability of the code.

Reply:
I had thought about this and originally was doing something like this. But, I'm not just validating the number here, I'm doing some fancy stuff to move the cursor around and change the input based on where the user input a number. So, to do it with a number means I'd have to do a bunch of mod division to get the value of certain digit/decimal places and honestly I think that would be less readable but definitely still open to other options on this one. I agree its not super readable atm but just struggling to think of a much better way.

Reply:
> This one simplified to `(\d)(\d)(.?)((\d{0,2})?)`

Simplified